pdc/apps/common/renderers.py

```diff
@@ -46,6 +50,105 @@
+
+
+def _serializer_field_link(text, field_name, base_name):
+    return '<a href="%s#field-%s">%s</a>' % (reverse(base_name + '-list'), field_name, text)
+
+
+def _serializer_field_data_to_string(field_data, parent_fields):
+    value = field_data.values.pop('value')
+    value = _serializer_data_to_string(value, parent_fields)
+    value = _serializer_field_attribute_to_string('value', value)
+
+    detail_items = []
+    for key in ['tags', 'help']:
+        if key in field_data.values:
+            detail_items.append(_serializer_field_attribute_to_string(key, field_data.values[key]))
+
+    if detail_items:
+        value += ' ' + _serializer_field_attribute_to_string('details', ' '.join(detail_items))
+
+    return value
+
+
+def _indented_text(text):
+    return text.replace('\n', '\n' + SERIALIZER_DATA_INDENTATION)
+
+
+def _serializer_field_to_string(items, container_format):
+    result = ','.join(['\n' + SERIALIZER_DATA_INDENTATION + item for item in items])
+    return container_format % (result + '\n')
+
+
+def _serializer_field_dict_to_string(field, parent_fields):
+    items = []
+    for field_name, field_data in sorted(field.iteritems()):
+        key = _serializer_field_attribute_to_string('name', '"%s"' % field_name)
+        parent_fields.append(field_name)
+        value = _serializer_data_to_string(field_data, parent_fields)
+        field_id = 'field-' + '__'.join(parent_fields)
+        parent_fields.pop()
+        html = '<span class="serializer-field" id="%s">%s: %s</span>' % (field_id, key, _indented_text(value))
+        items.append(html)
+
+    return _serializer_field_to_string(items, '{%s}')
+
+
+def _serializer_field_list_to_string(field, parent_fields):
+    items = []
+    for field_data in field:
+        value = _serializer_data_to_string(field_data, parent_fields)
+        items.append(_indented_text(value))
+
+    if len(items) == 1:
+        return '[ %s, &hellip; ]' % items[0]
+
+    return _serializer_field_to_string(items, '[%s]')
+
+
+def _model_field_reference_to_string(data):
+    match = MODEL_FIELD_REFERENCE_RE.match(data)
+    if match is not None:
+        model_name = match.group(1)
+        field_name = match.group(2)
+
+        for model_class, base_name in _models_and_base_names():
+            if model_class.__name__ == model_name:
+                return _serializer_field_link(data, field_name, base_name)
+
+    return data
+
+
+def _serializer_data_to_string(data, parent_fields=None):
+    if parent_fields is None:
+        parent_fields = []
+
+    if isinstance(data, _SerializerFieldData):
+        return _serializer_field_data_to_string(data, parent_fields)
+
+    if isinstance(data, dict):
+        return _serializer_field_dict_to_string(data, parent_fields)
+
+    if isinstance(data, list):
+        return _serializer_field_list_to_string(data, parent_fields)
+
+    if isinstance(data, str) or isinstance(data, unicode):
```

Comment:
This could be `isinstance(data, (str, unicode))`.

Reply:
Oh, I should test this against `six.string_types`.